### ISSUE TITLE
Renaming Data Sets

### ISSUE THREAD

**Issue by matheger** [veusz/veusz#431]
When renaming a data set that is in use somewhere, the expressions or plots that depend on it don't update with the new names and throw errors in the log window. (Side note: Those errors are unfortunately not very descriptive and don't tell you _where_ that error is happening, so you'll have to spend some time chasing them down.) The best fix would obviously be that renaming a data set in the Data Navigator window would propagate all name changes to wherever that data set is in use. 

At the moment this issue would be at the very top of my personal wish list on how to make an already awesome piece of software even better.

**Comment by jeremysanders** [comment]
It would be possible to change some of the uses, but not all (e.g. expressions involving the dataset name)

**Comment by matheger** [comment]
I see... I suppose it's because the user has to be able to reference data sets by name in order to enter expressions?

**Comment by jeremysanders** [comment]
Yes, e.g. my_data*2. There are other places datasets can appear within a document that can't be trivially be enumerated over. Maybe it would be possible to change some of these, but the inconsistency might be an issue.
